react/week4/gitrepos/src/components/Form.js

```diff
@@ -0,0 +1,20 @@
+import { useContext } from "react";
+import { GlobalStateContext } from "../contexts/GlobalStateContext";
+
+export default function Form() {
+  const { query, setQuery } = useContext(GlobalStateContext);
+  function handleQueryChange(e) {
+    setQuery(e.target.value);
+    console.log(e.target.value);
+  }
```

Comment:
When it's not manipulating any data and it's simply calling the setQuery function I would get in the habit of using it directly on the event handler.

If you do something with the parameter before setting then I would abstract as you've done here!